### DISCUSSION TITLE
Custom test macro is not evaluated as runnable

### DISCUSSION THREAD

**Discussion by MayankJikadara** [zed-industries/zed#15580] [Feature Requests]
I'm using a custom macro called `my_tests` which is a wrapper over the regular `#[test]` attribute, similar to the [test_log](https://docs.rs/test-log/0.2.10/src/test_log/lib.rs.html#176) crate. However, when I use this macro, the Zed editor (version 0.146.3) doesn't recognize the test function as runnable and doesn't show the "run" action next to the function.

I suspect that Zed is expecting the attribute value to be "test", as indicated by the [code reference]((https://github.com/zed-industries/zed/blob/a08e020e32cf20fad53f1f4122b979fd5dc99f25/crates/languages/src/rust/runnables.scm#L16)) in the Zed repository. The same code works fine in Visual Studio Code and shows the runnable actions.

How can I get Zed to recognize the expanded macro, which would have the `#[test]` attribute, so that it can properly identify the test functions as runnable?"

**Comment by SomeoneToIgnore** [comment]
> I suspect that Zed is expecting the attribute value to be "test", as indicated by the code reference in the Zed repository. 

Indeed, this is due https://github.com/zed-industries/zed/blob/c98918aed84a198447d73f6155f3c662f81e3053/crates/languages/src/rust/runnables.scm#L12-L17 and we might want to relax it to check whether an attribute contains the word "test" instead of the full match?

> The same code works fine in Visual Studio Code and shows the runnable actions.

VSCode has a much better introspection of tests, thanks to rust-analyzer macro expansions and the rest of the logic + its extension that's registering tasks based on those expansions.

I think the only way currently is to tweak the file above and make a Zed build with the new file, IIRC there's still no way to dynamically overload these files.

**Comment by MayankJikadara** [comment]
>  and we might want to relax it to check whether an attribute contains the word "test" instead of the full match?

That's a good stop gap solution.

**Reply by MayankJikadara** [reply]
https://github.com/zed-industries/zed/pull/15647

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
